### PR TITLE
added reserve configuration table and functionality

### PR DESCRIPTION
**Pull request by ASLouw**
currently cell size is fixed to 1 km

### CODE REVIEW

File: server/src/entity/reserve-configuration.entity.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Entity, PrimaryColumn ,Column } from 'typeorm';
```
Comment (EvansMatthew97) [inline server/src/entity/reserve-configuration.entity.ts]: Can you fix the comma here?

File: server/src/entity/reserve-configuration.entity.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+@Entity()
+export class ReserveConfiguration {
+  @PrimaryColumn('text')
+  reserveName: string;
+
+  @Column("int")
```
Comment (EvansMatthew97) [inline server/src/entity/reserve-configuration.entity.ts]: Use single quotes

File: server/src/services/map-updater.service.ts [hunk 3]
```diff
@@ -3,11 +3,16 @@ import { Injectable } from '@nestjs/common';
 import { OverpassService } from './overpass.service';
 import { MapPartitionerService } from './map-partitioner.service';
 
+import { DatabaseService } from './db.service';
+import { ReserveConfiguration } from '../entity/reserve-configuration.entity'
```
Comment (EvansMatthew97) [inline server/src/services/map-updater.service.ts]: Missing semicolon

**Review by EvansMatthew97 — CHANGES_REQUESTED** [review]
I think that the configuration implementation could be better achieved if you use a key-value table, rather than assigning the configuration keys to column names.

So for instance the table could be

```
name        | value
-----------------------------------
reserveName | "Kruger National Park"
cellSize    | 0.4
```

All name fields are strings, as are values. The values would be JSON.


This would mean we can add any new values to the config without having to redefine the database structure.

File: server/src/entity/reserve-configuration.entity.ts (hunk 1 above)
Comment (ASLouw) [reply server/src/entity/reserve-configuration.entity.ts]: Will do

File: server/src/entity/reserve-configuration.entity.ts (hunk 2 above)
Comment (ASLouw) [reply server/src/entity/reserve-configuration.entity.ts]: Right forgot about the standard

File: server/src/services/map-updater.service.ts (hunk 3 above)
Comment (ASLouw) [reply server/src/services/map-updater.service.ts]: Will fix

**Review by Viidas96 — COMMENTED** [review]
not sure about the functionality

**Review by EvansMatthew97 — APPROVED** [review]

**Pull request merged**